plugin/audit_log_filter/audit_log_reader.cc

```diff
@@ -223,15 +223,29 @@ bool AuditLogReader::read(AuditLogReaderContext *reader_context) noexcept {
               reader_context->current_file)) {
         return false;
       }
-    }
 
-    rapidjson::Reader reader;
-    reader.IterativeParseInit();
+      if (reader_context->reader == nullptr) {
+        reader_context->reader = std::make_unique<rapidjson::Reader>();
+      }
 
-    while (!reader.IterativeParseComplete()) {
-      reader.IterativeParseNext<rapidjson::kParseDefaultFlags>(
+      reader_context->reader->IterativeParseInit();
+    } else if (reader_context->reader == nullptr) {
+      reader_context->reader = std::make_unique<rapidjson::Reader>();
```

Comment:
Could you please explain when such situation is possible? I.e. when we can have reader_context->current_file set but reader not created/initialized... I think it is worth adding a comment at very least.